### ISSUE TITLE
Formatting messes up cursor position

### ISSUE THREAD

**Issue by DrunkenToast** [nvimdev/guard.nvim#69]
Sometimes on format the cursor will move to the top of the file.
I have not been able to properly reproduce this.

Used config:
```lua
    {
        "nvimdev/guard.nvim",
        event = { 'BufReadPost', 'BufNewFile' },
        config = function()
            local ft = require('guard.filetype')

            ft('typescript,javascript,typescriptreact,vue'):fmt('prettierd')
            --     :lint({
            --     cmd = 'eslint',
            --     args = { '--fix' },
            --     stdin = true
            -- })

            -- Call setup() LAST!
            require('guard').setup({
                -- the only options for the setup function
                fmt_on_save = true,
                -- Use lsp if no formatter was defined for this filetype
                lsp_as_default_formatter = false,
            })
        end
    },
```

Possibly related: #68

**Comment by barrett-ruth** [comment]
Should involve handling exit code of uv.spawn() process for formatters

I suggest we also create a config option `check_exit_code` like null-ls. Otherwise check if exit code is 0.

**Comment by barrett-ruth** [comment]
Can you provide an example, try to reproduce it, or provide any helpful info?

Also BTW, that lint config with eslint (which may actually be the problem - are you sure it's the formatte) will not work.
We need to manually parse the output of linters and haven't implemented eslint nor eslint_d yet (those should be coming very soon, though).

**Comment by DrunkenToast** [comment]
> Can you provide an example, try to reproduce it, or provide any helpful info?

I will update once I am able to reproduce the issue but so far it seems to happen on random occasions.
The current workaround has always been to completely restart nvim because if it happens once, it keeps happening.

**Comment by glepnir** [comment]
if you got something wrong. I'd suggest you try a min config avoid some things conflict.

**Comment by DrunkenToast** [comment]
> if you got something wrong. I'd suggest you try a min config avoid some things conflict.

Config is posted above. It is already very minimal.

**Comment by glepnir** [comment]
I mean just include guard . something like  this not a **Config**

```lua
vim.opt.rtp:append(path/to/guard)
local ft = require('guard.filetype')
ft:('xxx'):fmt
```

**Comment by barrett-ruth** [comment]
@DrunkenToast also are you on latest version of guard, pretty sure this was fixed in #48

**Comment by DrunkenToast** [comment]
> @DrunkenToast also are you on latest version of guard, pretty sure this was fixed in #48 

I regularly update plugins, it did happen again recently in a vue3 project. But again, I don't really understand what causes it.

**Comment by DrunkenToast** [comment]
The issue has not occurred anymore since the exit code fix. The only thing that annoys me currently is that undoing makes my cursor jump to the bottom and with prettier it fails very very often even if there is no reason to (which I assume means it sends the wrong buffer as input), but these are separate issues so I'll consider this closer for now.
